### ISSUE TITLE
Getting Started Documentation with MongoDB on Prisma should talk about Replica Set requirement (Epic 3)

### ISSUE THREAD

**Issue by matthewmueller** [prisma/prisma2-docs#2795]
~Currently our getting started guide assumes you know how to get a MongoDB connection URL _from somewhere_:~
- Update: we'll do the above in the Prisma Data Guide because getting a connection URL is not only relevant to Prisma Users.

https://www.prisma.io/docs/getting-started/setup-prisma/start-from-scratch/mongodb/connect-your-database-typescript-mongodb

This assumes that the MongoDB url is configured as a replica set. The default local MongoDB does not work like this and you'll end up with an error like this:

```
Transactions are not supported by this deployment
```

There's a bunch of good suggestions in [this issue](https://github.com/prisma/prisma/issues/8266) on how to configure your local MongoDB to support the replicaset.

Extra: consider also updating Getting Started Guide example schema to use embedded types.

**Comment by matthewmueller** [comment]
Update, we'll probably move this to the Prisma Data Guide. See [this discussion](https://prisma-company.slack.com/archives/CFM8D5VHC/p1643660083691719)

**Comment by Jolg42** [comment]
I think we should document the current MongoDB docker compose setup Pierre created, it might be the easiest one command setup (if you have docker compose).

https://github.com/prisma/prisma/blob/main/docker/docker-compose.yml#L91-L99
So 
```docker-compose.yml
version: '3.7'

services:
  mongodb:
    build: ./mongodb-replica
    environment:
      MONGO_INITDB_ROOT_USERNAME: root
      MONGO_INITDB_ROOT_PASSWORD: prisma
      MONGO_REPLICA_HOST: localhost
      MONGO_REPLICA_PORT: 27017
    ports:
      - '27017:27017'
```
Note I changed the port from current setup to use the default port (so we'll need to do a quick check if that still works after, it should, but let's verify)

https://github.com/prisma/prisma/blob/main/docker/mongodb_replica/Dockerfile
```Dockerfile
FROM mongo:5

# we take over the default & start mongo in replica set mode in a background task
ENTRYPOINT mongod --port $MONGO_REPLICA_PORT --replSet rs0 --bind_ip 0.0.0.0 & MONGOD_PID=$!; \
# we prepare the replica set with a single node and prepare the root user config
INIT_REPL_CMD="rs.initiate({ _id: 'rs0', members: [{ _id: 0, host: '$MONGO_REPLICA_HOST:$MONGO_REPLICA_PORT' }] })"; \
INIT_USER_CMD="db.createUser({ user: '$MONGO_INITDB_ROOT_USERNAME', pwd: '$MONGO_INITDB_ROOT_PASSWORD', roles: [ 'root' ] })"; \
# we wait for the replica set to be ready and then submit the commands just above
until (mongo admin --port $MONGO_REPLICA_PORT --eval "$INIT_REPL_CMD && $INIT_USER_CMD"); do sleep 1; done; \
# we are done but we keep the container by waiting on signals from the mongo task
echo "REPLICA SET ONLINE"; wait $MONGOD_PID;
```

Note that this setup is only recommended for development.

**Comment by paulrostorp** [comment]
@Jolg42 I tweaked the dockerfile to get it to work on my machine
1. switch to use `bind_ip_all`
2. (optional) I also disabled authentication, so that databases can be created on the fly. I find this useful when running several services locally, each using their own database.
```dockerfile
FROM mongo:5

# we take over the default & start mongo in replica set mode in a background task
ENTRYPOINT mongod --port $MONGO_REPLICA_PORT --replSet rs0 --bind_ip_all & MONGOD_PID=$!; \
# we prepare the replica set with a single node and prepare the root user config
INIT_REPL_CMD="rs.initiate()"; \
# we wait for the replica set to be ready and then submit the commands just above
until (mongo admin --port $MONGO_REPLICA_PORT --eval "$INIT_REPL_CMD"); do sleep 1; done; \
# we are done but we keep the container by waiting on signals from the mongo task
echo "REPLICA SET ONLINE"; wait $MONGOD_PID;
```

An additional requirement when using replicaSet and no auth is to use the `directConnection` flag

```
DATABASE_URL="mongodb://localhost:27017/my-service?directConnection=true"
```

**Comment by matthewmueller** [comment]
I'd also like us to cover the non-docker way, 

On macOS, there's some fumbling by me in this issue with @Jolg42 coming in to rescue me 🦸  

https://github.com/prisma/prisma/issues/11860#issuecomment-1043863879

**Comment by matthewmueller** [comment]
Update: I've written up the 3 different options the user can use to get a MongoDB database that supports replica sets. 

https://github.com/prisma/prisma/issues/8713#issuecomment-1045268231

**Comment by garrensmith** [comment]
I've added in a pris.ly link for this error message https://github.com/prisma/pris.ly/pull/41: https://pris.ly/d/mongodb-replica-set
This link will have to be updated to point to the correct doc's page once it has been created.

**Comment by fullStackDataSolutions** [comment]
Is there a docker compose file in existence that I can use to set this up? The link above 404s.

**Comment by matthewmueller** [comment]
@blazestudios23 thanks for the heads up, I just updated the link above

**Comment by matthewmueller** [comment]
Notes:
- Must have: document using Atlas  in getting started
- Nice to have: Ideally, offer a local alternative.

**Comment by matthewmueller** [comment]
Closed by: https://github.com/prisma/docs/pull/3017

**Comment by Jolg42** [comment]
I created a separate issue to add documentation for docker-compose 
https://github.com/prisma/docs/issues/3040

**Comment by samrocksc** [comment]
this is literally the biggest barrier to entry for a premium level system i've ever used.. I've used Prisma for Postgres, and it was smooth...but this is incredibly hard for me to follow.
